frontend/src/app/main/ui/workspace/sidebar/options/menus/fill.cljs

```diff
@@ -27,22 +32,60 @@
 (def fill-attrs-shape
   (conj fill-attrs :hide-fill-on-export))
 
+(defn color-values
+  [color]
+  {:color (:fill-color color)
+   :opacity (:fill-opacity color)
+   :id (:fill-color-ref-id color)
+   :file-id (:fill-color-ref-file color)
+   :gradient (:fill-color-gradient color)})
+
 (mf/defc fill-menu
   {::mf/wrap [#(mf/memo' % (mf/check-props ["ids" "values"]))]}
   [{:keys [ids type values disable-remove?] :as props}]
-  (let [show? (or (not (nil? (:fill-color values)))
-                  (not (nil? (:fill-color-gradient values))))
-
-        label (case type
+  (let [label (case type
                 :multiple (tr "workspace.options.selection-fill")
                 :group (tr "workspace.options.group-fill")
                 (tr "workspace.options.fill"))
 
-        color {:color (:fill-color values)
-               :opacity (:fill-opacity values)
-               :id (:fill-color-ref-id values)
-               :file-id (:fill-color-ref-file values)
-               :gradient (:fill-color-gradient values)}
+        ;; Excluding nil values
+        values (->> values
+                    (remove (comp nil? val))
+                    (into {}))
+
+        only-shapes? (and (contains? values :fills)
+                          ;; texts have :fill-* attributes, the rest of the shapes have :fills
+                          (= (count (filter #(str/starts-with? (str %) ":fill-") (keys values))) 0))
```

Comment:
To convert a key into a string is better to use `name`. Also I think it's better to move this into a function

Reply:
Thanks!, it's using (name %) "fill-" now

About moving it into a function, I think this code will disappear as soon as the new feature about texts is merged (the will be considered as a normal shape for this case so the check won't be necessary)